files/grest/rpc/transactions/tx_info.sql

```diff
@@ -644,52 +644,89 @@ BEGIN
+              WHERE redeemer.tx_id = ANY (_tx_id_list)
+            )
           SELECT
-            redeemer.tx_id,
+            ar.tx_id,
             JSONB_BUILD_OBJECT(
-              'address', INUTXO.address,
-              'script_hash', ENCODE(script.hash, 'hex'),
-              'bytecode', ENCODE(script.bytes, 'hex'),
-              'size', script.serialised_size,
-              'valid_contract', tx.valid_contract,
+              'address', 
+                CASE 
+                  WHEN ar.purpose = 'spend' THEN
+                    (SELECT address FROM spend_redeemers sr WHERE sr.id = ar.id)
+                  ELSE NULL
+                END,
+              'script_hash', ENCODE(ar.script_hash, 'hex'),
+              'bytecode', ENCODE(ar.script_bytes, 'hex'),
+              'size', ar.script_serialised_size,
+              'valid_contract', ar.valid_contract,
               'input', JSONB_BUILD_OBJECT(
                 'redeemer', JSONB_BUILD_OBJECT(
-                  'purpose', redeemer.purpose,
-                  'fee', redeemer.fee::text,
+                  'purpose', ar.purpose,
+                  'fee', ar.fee::text,
                   'unit', JSONB_BUILD_OBJECT(
-                    'steps', redeemer.unit_steps::text,
-                    'mem', redeemer.unit_mem::text
+                    'steps', ar.unit_steps::text,
+                    'mem', ar.unit_mem::text
                   ),
                   'datum', JSONB_BUILD_OBJECT(
-                    'hash', ENCODE(rd.hash, 'hex'),
-                    'value', rd.value
+                    'hash', ENCODE(ar.rd_hash, 'hex'),
+                    'value', ar.rd_value
                   )
                 ),
-                'datum', JSONB_BUILD_OBJECT(
-                  'hash', ENCODE(ind.hash, 'hex'),
-                  'value', ind.value
-                )
+                'datum', CASE WHEN ar.purpose = 'spend' THEN (
+                    SELECT JSONB_BUILD_OBJECT(
+                      'hash', ENCODE(sr.ind_hash, 'hex'),
+                      'value', sr.ind_value
+                    ) FROM spend_redeemers sr WHERE sr.id = ar.id
+                  ) ELSE NULL END
               ),
-              'output', CASE WHEN outd.hash IS NULL THEN NULL
-                        ELSE
-                          JSONB_BUILD_OBJECT(
-                            'hash', ENCODE(outd.hash, 'hex'),
-                            'value', outd.value
-                          )
-                        END
+              'output', CASE WHEN ar.purpose = 'spend' THEN (
+                  SELECT 
+                    CASE WHEN sr.outd_hash IS NULL THEN NULL ELSE 
+                      JSONB_BUILD_OBJECT(
+                        'hash', ENCODE(sr.outd_hash, 'hex'),
+                        'value', sr.outd_value
+                      )
+                    END
+                  FROM spend_redeemers sr WHERE sr.id = ar.id
+                ) ELSE NULL END
```

Comment:
Remove from contract as there is no actual link between spent input (what the spend contract is for) and any potential output on the same contract address.